app/src/components/Layout/Box.js

```diff
@@ -0,0 +1,109 @@
+import { useState, Fragment, useRef } from "react";
+
+import Form from "../Forms/Form";
+import Button from "../UI/Button";
+import ErrorModal from "../UI/ErrorModal";
+import { search } from "../../CreateExcelFile/GoogleSearch";
+import { createExcel } from "../../CreateExcelFile/ExcelUtils";
+import "./Box.css";
+
+function Box() {
+  const queryInputRef = useRef();
+  const numberOfResultsInputRef = useRef();
+  const [showForm, setShowForm] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [errorText, setErrorText] = useState(false);
+
+  const onSubmitHandler = async (event) => {
+    event.preventDefault();
+    let queryInput = queryInputRef.current.value.trim();
+    let numberOfResultsInput = numberOfResultsInputRef.current.value;
+
+    if (
+      queryInput.length > 0 &&
+      numberOfResultsInput >= 1 &&
+      numberOfResultsInput <= 99
+    ) {
+      setIsLoading(true);
+      try {
+        const googleResult = await search(queryInput, numberOfResultsInput); 
+
+        createExcel(googleResult, queryInput);
+
+      } catch {
+        setError(true);
+        setErrorText("Could not fetch searches");
+      }
+      setIsLoading(false);
+      setShowForm(false);
+  
+    } else if (queryInput.length < 1) {
+      setError(true);
+      setErrorText("To fetch searches you need to add a query");
+    
+    } else if (numberOfResultsInput < 1 || numberOfResultsInput > 99) {
+      setError(true);
+      setErrorText("Please enter a search number between 1 and 99");
+    }
+  };
+
+  const onDownloadHandler = () => {}; //? may use later on?
```

Comment:
do you not plan to implement this? the download manually option

Reply:
Think I said it bfr, wanna add this later cz I cant really figure it out...